### ISSUE TITLE
How to get contract storage struct?

### ISSUE THREAD

**Issue by earthican** [ethereum/web3.js#1728]
I am using Web3 v1.0.0-beta.34, and I'm wondering how to get struct data from my contract. Example:

```solidity
contract MyContract {
...
  struct MyStruct {
    bytes data;
  }
  MyStruct[100] public structs;
  constructor() {
    structs[0] = new MyStruct("some data");
  }
...
}
```

In Web3, I make the call:
```javascript
let abi = require('contract-abi.json').abi;
let address = "/* contract address */";
let contract = new web3.eth.Contract(abi, address);

contract.methods.structs(0).call().then(console.log);
// logs a hash, like "0x746869732069732..."
```

I was expecting the return of the call to `structs(0)` to be a JSON format of a MyStruct object. Instead, I'm getting a hash. What is this hash, and can I use it to retrieve data from the struct?

**Comment by davinci26** [comment]
I had the same issue and I found this
https://ethereum.stackexchange.com/questions/3609/returning-a-struct-and-reading-via-web3

It basically says that you cant return a struct.

**Comment by earthican** [comment]
Thanks for the find @davinci26. This makes sense, since the solidity compiler won't even let create a function that returns a struct (unless you're using experimental features).

But I'm not creating a getter function, only accessing the storage state variable `structs`. I was following the [CryptoZombies](https://cryptozombies.io/en/lesson/6/chapter/4) tutorial, and in the linked chapter they demonstrated this same method call and how would return as a JSON.

I was curious so I did some more digging. Here's what I found...

First interesting thing I noticed is that the hex string that was returning from my call to `structs(0)` is actually the `MyStruct.data` encoded as a hex string, so I was able to get the data, just not in JSON format.

Turns out that because there is only one property declared in `MyStruct`, web3 would return just the value of that one property.

When I added a second property (e.g. `string message`), web3 would then return an list of values:`["0x746869732069732...", "my message"]`.

So, we can still access public `struct` accessors, but without the proper data structure. Instead it just returns an order list with just the data associated within it.

BTW, I'm using web3 v1.0.0-beta.34 with Ganache v1.1.0 as the development blockchain.

This is my first time working on a dapp so I've never used any other version of web3 before. I'm wondering if the data structure of these return values changed between the different beta versions of web3?
